### ISSUE TITLE
40mm projectile entities drop out of muzzle

### ISSUE THREAD

**Issue by MrFroglegs** [CurlySparkle/ARC9-COD2019#107]
When firing an UBGL or the MGL-32, the projectile entities will initially drop out of the muzzle, then go to their true position after a couple of seconds. I am using the latest github commits for both this and the shared attachments repositories as of this post. This occurs in both singleplayer and multiplayer.

https://github.com/user-attachments/assets/c9b22b15-e433-4c89-9096-6e529d05e7ee

**Comment by CurlySparkle** [comment]
does this happens with sandbox gamemode or a different one? I got told this seems to happen on the darkrp gamemode

**Comment by MrFroglegs** [comment]
I've only tested it on sandbox so far, but I can test it on my friend's deathmatch gamemode to see if it happens on that as well.

**Comment by CurlySparkle** [comment]
be sure to see if this only happens with this type only, if it doesn't happen to the other launchers then It will narrow down the problem

**Comment by MrFroglegs** [comment]
Tested on my friend's deathmatch gamemode and on sandbox again and it still happens and it's only with the UBGLs and MGL-32. The rocket launchers do not suffer with this problem.

**Comment by CurlySparkle** [comment]
I see, im gonna push an update to see if that fixes the problem, let me know

**Comment by MrFroglegs** [comment]
Tested on both singleplayer and multiplayer in both sandbox and my friend's deathmatch gamemode, and it appears to work now. However, I did spot a couple of other little issues while I was testing. These happen in both single and multiplayer on sandbox and the DM gamemode:

Seems to be a particle rending issue. The orange glow renders over world geometry and props. Happens with the 40mm projectiles, as well as the PILA and RPG rockets that share the same particle it seems:
https://github.com/user-attachments/assets/2bcbab93-5db9-4b23-a674-92acb878060f

Additional projectile entities occasionally spawn when you fire too close to props or world geometry for the projectile to detonate. It appears random with how much it spawns - sometimes it's just the one shot (as it normally should be), and other times it spawns an additional 1-4 models:
https://github.com/user-attachments/assets/00ee0fac-1d33-4e1a-98e8-5db9b7ce280e
